app/controllers/comment_controller.php

```diff
@@ -0,0 +1,30 @@
+<?php 
+Class CommentController extends AppController
+{
+    public function mostLiked()
+    {
+        $comments = Comment::get_most_liked();
+        $this->set(get_defined_vars());
+    }
+
+    public function delete()
+    {
+        $comment = Comment::get(Param::get('comment_id'));
+        $comment->delete();
+        redirect(url('thread/view', array('thread_id' =>$_SESSION['thread_id'])));
```

Comment:
Please fix spacing 
```php
redirect(url('thread/view', array('thread_id' => $_SESSION['thread_id'])));
```
also you can create a separate model that handles session (with setter and getter methods)
that way, you would not be repeatedly use the $_SESSION global variable but instead just get it's value through the getter
and set a new value through the setter